plugins/kotlin/gradle/gradle-java/src/org/jetbrains/kotlin/idea/gradleJava/configuration/KotlinGradleCoroutineDebugProjectResolver.kt

```diff
@@ -29,6 +33,12 @@ class KotlinGradleCoroutineDebugProjectResolver : AbstractProjectResolverExtensi
             gradle.taskGraph.whenReady { TaskExecutionGraph taskGraph ->
                 taskGraph.allTasks.each { Task task ->
                     if (!(task instanceof Test || task instanceof JavaExec)) return
+                    for (arg in task.getAllJvmArgs() + task.getJvmArgs() + task.getArgs()) {
```

Comment:
Could we just check `System.getProperty(KOTLINX_COROUTINES_DEBUG_PROPERTY_NAME) == KOTLINX_COROUTINES_DEBUG_OFF_VALUE`?

Why do we need to run through all the task arguments?

Reply:
Fetching `System.getProperty(KOTLINX_COROUTINES_DEBUG_PROPERTY_NAME)` will fetch the property for the gradle VM, not the target one. Another this is that the property we are looking for can be returned by one of the args methods depending on how it was added to the task. For example if we add this block to the build.gradle file:
```
application {
    mainClassName = 'MainKt'
    applicationDefaultJvmArgs = ['-Dkotlinx.coroutines.debug=off']
}
```
The property will be returned by `getJvmArgs()`.
For this block:
```
test {
    useJUnitPlatform()
    jvmArgs '-Dkotlinx.coroutines.debug=off'
}
```
The property will be returned by `getAllJvmArgs()`, but not by `getJvmArgs()`

I actually found out that `getArgs()` is not needed, so I removed it.

Reply:
Ok, thank you for the clarification